addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -1099,21 +1105,23 @@ var SnippetEditor = Widget.extend({
         // Remove the siblings that belong to a snippet in grid mode
         // and put the identified grid mode snippets in their own "selector".
         const selectorGrids = new Set();
-        if (rowEl.classList.contains('row')) {
-            if ($selectorSiblings) {
-                // Looping backwards because elements are removed, so the
-                // indexes are not lost.
-                for (let i = $selectorSiblings.length - 1; i >= 0; i--) {
-                    if (isPopup && !$selectorSiblings[i].closest('div.s_popup')) {
-                        // Removing the siblings that are outside the popup if
-                        // the grid item is in a popup.
+        if (rowEl.classList.contains('row') && $selectorSiblings) {
+            // The grid from which the dragged column comes (if any).
+            const targetGridEl = rowEl.classList.contains('o_grid_mode') ? rowEl : rowEl.closest('.o_grid_mode');
+            // Looping backwards because elements are removed, so the
+            // indexes are not lost.
+            for (let i = $selectorSiblings.length - 1; i >= 0; i--) {
+                if (isPopup && !$selectorSiblings[i].closest('div.s_popup')) {
+                    // Removing the siblings that are outside the popup if
+                    // the grid item is in a popup.
+                    $selectorSiblings.splice(i, 1);
+                } else {
+                    const gridSnippetEl = $selectorSiblings[i].closest('.o_grid_mode');
+                    // Keep the inner siblings when dragging an inner snippet
+                    // column that is in a grid.
+                    if (gridSnippetEl && (gridSnippetEl !== targetGridEl || gridSnippetEl === rowEl)) {
```

Comment:
I think you can improve comment on that whole section because I am still not entirely sure to understand the fix :wink: It works well but... :wink: